csource/common.go

```diff
@@ -106,6 +114,79 @@ static void install_segv_handler()
 	sigaction(SIGBUS, &sa, NULL);
 }
 
+#ifdef __NR_syz_emit_ethernet
+#define COMMAND_MAX_LEN 64
+
+static void execute_command(const char *format, ...)
+{
+	va_list args;
+	int rv;
+	char command[COMMAND_MAX_LEN];
+
+	va_start(args, format);
+
+	rv = vsnprintf(&command[0], sizeof(command), format, args);
+	if (rv < 0) {
+		perror("snprintf()");
+		exitf("tun: snprintf failed");
+	}
+	if ((unsigned)rv >= sizeof(command)) {
+		fail("tun: command \"%s\"doesn't fit into buffer", &command[0]);
+	}
+	if (system(&command[0]) < 0) {
+		perror(&command[0]);
+		exitf("tun: command \"%s\" failed", &command[0]);
+	}
+
+	va_end(args);
+}
+
+int tunfd;
+
+#define LOCAL_MAC "42:42:42:42:42:42"
+#define LOCAL_IPV4 "192.168.1.5"
+#define LOCAL_IPV6 "fe80::4042:42ff:fe42:4242"
+#define REMOTE_MAC "aa:aa:aa:aa:aa:aa"
+#define REMOTE_IPV4 "192.168.1.6"
+
+static void initialize_tun()
+{
+	int rv;
+
+	if (getuid() != 0)
+		return;
+
+	tunfd = open("/dev/net/tun", O_RDWR);
+	if (tunfd == -1) {
+		perror("open()");
+		exitf("tun: can't open /dev/net/tun");
+	}
+	char name[IFNAMSIZ];
+	rv = snprintf(&name[0], sizeof(name), "syz%d", getpid());
+	if (rv < 0) {
+		perror("snprintf()");
+		exitf("tun: snprintf failed");
+	}
+	assert((unsigned)rv < sizeof(name));
+	name[sizeof(name) - 1] = '\0';
+
+	struct ifreq ifr;
+	memset(&ifr, 0, sizeof(ifr));
+	strncpy(ifr.ifr_name, &name[0], IFNAMSIZ);
+	ifr.ifr_flags = IFF_TAP | IFF_NO_PI;
+	if (ioctl(tunfd, TUNSETIFF, (void *)&ifr) < 0) {
+		perror("ioctl(TUNSETIFF)");
+		fail("tun: ioctl(TUNSETIFF) failed");
+	}
+
+	execute_command("ip link set dev %s address %s", &name[0], LOCAL_MAC);
+	execute_command("ip addr add %s/24 dev %s", LOCAL_IPV4, &name[0]);
+	execute_command("ip -6 addr add %s/64 dev %s", LOCAL_IPV6, &name[0]);
+	execute_command("ip link set %s up", &name[0]);
+	execute_command("arp -s %s %s", REMOTE_IPV4, REMOTE_MAC);
+}
+#endif /* __NR_syz_emit_ethernet */
```

Comment:
The C style comments are not stripped by prog2c, and executor does not use C style comments. Use //